src/custom-apps/components/WebhookHeaders/WebhookHeadersTableBody.tsx

```diff
@@ -0,0 +1,118 @@
+import TableRowLink from "@dashboard/components/TableRowLink";
+import { FormChange } from "@dashboard/hooks/useForm";
+import { removeAtIndex, updateAtIndex } from "@dashboard/utils/lists";
+import { TableBody, TableCell, TextField } from "@material-ui/core";
+import { DeleteIcon, IconButton } from "@saleor/macaw-ui";
+import clsx from "clsx";
+import React, { ChangeEvent } from "react";
+import { useIntl } from "react-intl";
+
+import { messages } from "./messages";
+import useStyles from "./styles";
+import { Header, stringifyHeaders } from "./utils";
+
+const nameSeparator = ":";
+const nameInputPrefix = "name";
+const valueInputPrefix = "value";
+
+export interface WebhookHeadersTableBodyProps {
+  onChange: FormChange;
+  headers: Header[];
+}
+
+const WebhookHeadersTableBody: React.FC<WebhookHeadersTableBodyProps> = ({
+  onChange,
+  headers,
+}) => {
+  const classes = useStyles();
+  const intl = useIntl();
+
+  const change = ({ target }: ChangeEvent<HTMLTextAreaElement>) => {
+    const { name, value } = target;
+    const [field, index] = name.split(nameSeparator);
+
+    const item = headers[index];
+
+    // lowercase header name
+    if (field === nameInputPrefix) {
+      item[field] = value.toLowerCase();
+    } else {
+      item[field] = value;
+    }
```

Comment:
**suggestion**: it deserves a dedicated function.

Reply:
extracted